### PR TITLE
Backport support for Alpine Linux/musl libc

### PR DESCRIPTION
**Pull request by calderast**
Apply some of [Alpine's patches](https://git.alpinelinux.org/aports/tree/community/openjdk8), changes from the [original commit](https://github.com/openjdk/jdk/commit/63009f90ec7df053c936226663b463b8584364a9), and additional fixes for Alpine Linux build support on jdk8u. 

This PR is identical to https://github.com/adoptium/jdk8u/pull/8 (excluding whitespace changes) with a cleaner commit history.

### CODE REVIEW

**Review by gdams — APPROVED** [review]

**Comment by calderast** [comment]
New pipeline run against this branch for Alpine Linux (https://ci.adoptopenjdk.net/job/jdk8-alpine-linux-x64-hotspot/59/)

From #8:
Pipeline run for all other platforms (https://ci.adoptopenjdk.net/job/build-scripts/job/openjdk8-pipeline/2015/)
The above run failed on Windows x64 due to a pipeline issue, passing Windows x64 re-run here (https://ci.adoptopenjdk.net/job/build-scripts/job/openjdk8-pipeline/2017/)

**Comment by calderast** [comment]
One of the whitespace changes from the previous PR caused a separator error in sa.make - it's been fixed now, so this should be good to go

File: hotspot/make/linux/makefiles/sa.make [hunk 1]
```diff
@@ -59,9 +59,11 @@ SA_PROPERTIES = $(SA_CLASSDIR)/sa.properties
 
 # if $(AGENT_DIR) does not exist, we don't build SA
 # also, we don't build SA on Itanium or zero.
+# check for thread_db.h too (musl does not have it).
 
 all: 
-	if [ -d $(AGENT_DIR) -a "$(SRCARCH)" != "ia64" \
+if [ -d $(AGENT_DIR) -a -f /usr/include/thread_db.h \
```
Comment (jiekang) [inline hotspot/make/linux/makefiles/sa.make]: @calderast it's the indentation here for this makefile that needs to be fixed

Comment (calderast) [reply hotspot/make/linux/makefiles/sa.make]: Fixed, thank you!

**Review by karianna — APPROVED** [review]
LGTM, matches previous PR in source diff and is cleaner in commit messages.  Passes builds on all platforms and AQAVit & TCK

**Comment by gdams** [comment]
Final build pipeline against correct branch for sanity reasons: https://ci.adoptopenjdk.net/job/build-scripts/job/openjdk8-pipeline/2035/

**Pull request merged**